certbot/tests/util.py

```diff
@@ -158,3 +162,87 @@ def make_lineage(self, testfile):
                 line.replace('MAGICDIR', self.config_dir) for line in src)
 
     return conf_path
+
+
+def patch_get_utility(target='zope.component.getUtility'):
+    """Patch zope.component.getUtility to use a special mock IDisplay.
+
+    The mock IDisplay works like a regular mock object, except it also
+    also asserts that methods are called with valid arguments.
+
+    :param str target: path to patch
+
+    :returns: mock zope.component.getUtility
+    :rtype: mock.MagicMock
+
+    """
+    return mock.patch(target, new_callable=_create_get_utility_mock)
+
+
+class FreezableMock(object):
+    """Mock object with the ability to freeze attributes.
+
+    This class works like a regular mock.MagicMock object, except
+    attributes and behavior can be set and frozen so they cannot be
+    changed during tests.
+
+    If a func argument is provided to the constructor, this function
+    is called first when an instance of FreezableMock is called,
+    followed by the usual behavior defined by MagicMock. The return
+    value of func is ignored.
+
+    """
+    def __init__(self, frozen=False, func=None):
+        self._frozen_set = set() if frozen else set(('freeze',))
+        self._func = func
+        self._mock = mock.MagicMock()
+        self._frozen = frozen
+
+    def freeze(self):
+        """Freeze object preventing further changes."""
+        self._frozen = True
+
+    def __call__(self, *args, **kwargs):
+        if self._func is not None:
+            self._func(*args, **kwargs)
+        return self._mock(*args, **kwargs)
+
+    def __getattribute__(self, name):
+        if name == '_frozen':
+            try:
+                return object.__getattribute__(self, name)
+            except AttributeError:
+                return False
+        elif name == '_frozen_set' or name in self._frozen_set:
+            return object.__getattribute__(self, name)
+        else:
+            return getattr(object.__getattribute__(self, '_mock'), name)
+
+    def __setattr__(self, name, value):
+        if self._frozen:
+            return setattr(self._mock, name, value)
+        elif name != '_frozen_set':
+            self._frozen_set.add(name)
+        return object.__setattr__(self, name, value)
+
+
+def _create_get_utility_mock():
+    display = FreezableMock()
+    for name in interfaces.IDisplay.names():  # pylint: disable=no-member
+        if name != 'notification':
```

Comment:
Are we sure we want to exclude notification? What if an existing method that we somehow magically missed *really* wants to force a pause?